diff --git a/app/assets/css/gatesHomeScreen.css b/app/assets/css/gatesHomeScreen.css
--- a/app/assets/css/gatesHomeScreen.css
+++ b/app/assets/css/gatesHomeScreen.css
@@ -263,3 +263,9 @@
    max-width: 500px;
    margin: 0 auto 35px;
 }
+
+
+.ongoing-list {
+  background-color: #d9534f !important;
+  border-color: #d9534f !important;
+}
diff --git a/app/assets/js/bundle.js b/app/assets/js/bundle.js
--- a/app/assets/js/bundle.js
+++ b/app/assets/js/bundle.js
@@ -10946,6 +10946,12 @@
   var $activeList = $('#active-species');
   var $completedList = $('#finished-species');
 
+  if (activeSpecies.length === 0) {
+    $activeList.removeClass('ongoing-list');
+  } else {
+    $activeList.addClass('ongoing-list');
+  }
+
   exports.addSpeciesToList(activeSpecies, $activeList);
   exports.addSpeciesToList(completedSpecies, $completedList);
 };
@@ -10975,6 +10981,12 @@
   var $activeList = $('#active-food');
   var $completedList = $('#finished-food');
 
+  if (activeFood.length === 0) {
+    $activeList.removeClass('ongoing-list');
+  } else {
+    $activeList.addClass('ongoing-list');
+  }
+
   exports.addFoodToList(activeFood, $activeList);
   exports.addFoodToList(completedFood, $completedList);
 };
diff --git a/app/assets/js/jgiFollow.js b/app/assets/js/jgiFollow.js
--- a/app/assets/js/jgiFollow.js
+++ b/app/assets/js/jgiFollow.js
@@ -290,6 +290,12 @@
   var $activeList = $('#active-species');
   var $completedList = $('#finished-species');
 
+  if (activeSpecies.length === 0) {
+    $activeList.removeClass('ongoing-list');
+  } else {
+    $activeList.addClass('ongoing-list');
+  }
+
   exports.addSpeciesToList(activeSpecies, $activeList);
   exports.addSpeciesToList(completedSpecies, $completedList);
 };
@@ -319,6 +325,12 @@
   var $activeList = $('#active-food');
   var $completedList = $('#finished-food');
 
+  if (activeFood.length === 0) {
+    $activeList.removeClass('ongoing-list');
+  } else {
+    $activeList.addClass('ongoing-list');
+  }
+
   exports.addFoodToList(activeFood, $activeList);
   exports.addFoodToList(completedFood, $completedList);
 };
